Makefile

```diff
@@ -10,32 +10,32 @@ EXAMPLES_BUILD_DIR = $(BUILD_DIR)/examples
 CC         = gcc
 CXX        = g++
 BUILD_CC   = gcc
-CFLAGS     = -Wall -Wextra -Wno-unused-parameter
+override CFLAGS += -Wall -Wextra -Wno-unused-parameter
 LDFLAGS    = -lm -ldl
 PHP        = php
 LLK        = llk
 #LLK        = $(PHP) $(HOME)/php/llk/llk.php
 
 ifeq (debug, $(BUILD))
- CFLAGS += -O0 -g -DIR_DEBUG=1
+ override CFLAGS += -O0 -g -DIR_DEBUG=1
 endif
```

Comment:
I've never known about the "override" before.
I think, it may be better to use a different BUILD type. e.g. "asan" in addition to "release" and "debug".

Reply:
https://www.gnu.org/software/make/manual/html_node/Override-Directive.html

> If a variable has been set with a command argument (see [Overriding Variables](https://www.gnu.org/software/make/manual/html_node/Overriding.html)), then ordinary assignments in the makefile are ignored. If you want to set the variable in the makefile even though it was set with a command argument, you can use an override directive, which is a line that looks like this:
>
> override variable = value

I think `override` is a good flag either way, because otherwise users can still set `CFLAGS` via `make CFLAGS=...` but local `CFLAGS` in the Makefile are completely ignored. This can lead to unexpected behavior. I.e. in this case it will cancel the `-Wall -Wextra -Wno-unused-parameter`, and `-O2 -g` with the `release` build, etc. This is not immediately obvious when the removal of the flags doesn't break the build.

Reply:
OK